Kernel/Interrupts/InterruptManagement.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <AK/FixedArray.h>
+#include <AK/Function.h>
+#include <AK/RefCounted.h>
+#include <AK/RefPtr.h>
+#include <AK/Types.h>
+#include <Kernel/ACPI/Definitions.h>
+#include <Kernel/Interrupts/GenericInterruptHandler.h>
+#include <Kernel/Interrupts/IOAPIC.h>
+#include <Kernel/Interrupts/IRQController.h>
+namespace Kernel {
+class InterruptManagement {
+public:
+    static InterruptManagement& the();
+    static void initialize();
+
+    static void switch_to_pic_mode();
+    static void switch_to_ioapic_mode();
+
+    virtual void enable(u8 interrupt_vector);
+    virtual void disable(u8 interrupt_vector);
+    virtual void eoi(u8 interrupt_vector);
+
+    void enumerate_interrupt_handlers(Function<void(GenericInterruptHandler&)>);
+
+    IRQController& get_interrupt_controller(int index);
+
+protected:
+    InterruptManagement(bool create_default_controller);
+    FixedArray<IRQController*> m_interrupt_controllers { 1 };
```

Comment:
Please remove all unnecessary use of `new`.
If you make this a `FixedArray<NonnullOwnPtr<IRQController>>`, you can create new entries for it using `make<IRQController>(...)` and they will not leak.